projects/storefrontlib/src/cms-structure/seo/page-meta-link.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable, Renderer2, RendererFactory2 } from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PageMetaLinkService {
+  constructor(
+    protected winRef: WindowRef,
+    protected rendererFactory: RendererFactory2
+  ) {}
+
+  /**
+   * Adds a canonical link element to the document head.
+   *
+   * If an id is provided, the link will be updated.
+   * If no url is provided, the link element will be deleted.
+   */
+  setCanonicalLink(url: string | undefined): void {
+    let link: HTMLLinkElement = document.querySelector(
```

Comment:
[medium] Using global `documet` throws an error in SSR
```suggestion
    let link: HTMLLinkElement = this.winRef.document.querySelector(
```